lib/ansible/module_utils/urls.py

```diff
@@ -976,6 +979,137 @@ def atexit_remove_file(filename):
+        context.options |= ssl.OP_NO_SSLv3
+        context.check_hostname = False
+        context.verify_mode = ssl.CERT_NONE
+
+    if validate_certs and any((cafile, cadata)):
+        context.load_verify_locations(cafile=cafile, cadata=cadata)
+
+    if ciphers:
+        context.set_ciphers(':'.join(map(to_native, ciphers)))
+
+    return context
+
+
+def get_ca_certs(cafile=None):
+    # tries to find a valid CA cert in one of the
+    # standard locations for the current distribution
+
+    cadata = bytearray()
+    paths_checked = []
+
+    if cafile:
+        paths_checked = [cafile]
+        with open(to_bytes(cafile, errors='surrogate_or_strict'), 'rb') as f:
+            if HAS_SSLCONTEXT:
+                for b_pem in extract_pem_certs(f.read()):
+                    cadata.extend(
+                        ssl.PEM_cert_to_DER_cert(
+                            to_native(b_pem, errors='surrogate_or_strict')
+                        )
+                    )
+        return cafile, cadata, paths_checked
+
+    if not HAS_SSLCONTEXT:
+        paths_checked.append('/etc/ssl/certs')
+
+    system = to_text(platform.system(), errors='surrogate_or_strict')
+    # build a list of paths to check for .crt/.pem files
+    # based on the platform type
+    if system == u'Linux':
+        paths_checked.append('/etc/pki/ca-trust/extracted/pem')
+        paths_checked.append('/etc/pki/tls/certs')
+        paths_checked.append('/usr/share/ca-certificates/cacert.org')
+    elif system == u'FreeBSD':
+        paths_checked.append('/usr/local/share/certs')
+    elif system == u'OpenBSD':
+        paths_checked.append('/etc/ssl')
+    elif system == u'NetBSD':
+        paths_checked.append('/etc/openssl/certs')
+    elif system == u'SunOS':
+        paths_checked.append('/opt/local/etc/openssl/certs')
+    elif system == u'AIX':
+        paths_checked.append('/var/ssl/certs')
+        paths_checked.append('/opt/freeware/etc/ssl/certs')
+
+    # fall back to a user-deployed cert in a standard
+    # location if the OS platform one is not available
+    paths_checked.append('/etc/ansible')
+
+    tmp_path = None
+    if not HAS_SSLCONTEXT:
+        tmp_fd, tmp_path = tempfile.mkstemp()
+        atexit.register(atexit_remove_file, tmp_path)
+
+    # Write the dummy ca cert if we are running on macOS
+    if system == u'Darwin':
+        if HAS_SSLCONTEXT:
+            cadata.extend(
+                ssl.PEM_cert_to_DER_cert(
+                    to_native(b_DUMMY_CA_CERT, errors='surrogate_or_strict')
+                )
+            )
+        else:
+            os.write(tmp_fd, b_DUMMY_CA_CERT)
+        # Default Homebrew path for OpenSSL certs
+        paths_checked.append('/usr/local/etc/openssl')
+
+    # for all of the paths, find any  .crt or .pem files
+    # and compile them into single temp file for use
+    # in the ssl check to speed up the test
+    for path in paths_checked:
+        if os.path.exists(path) and os.path.isdir(path):
```

Comment:
Removing some levels of indentation by checking the inverse of the current `if` statements + `continue` could make this section more readable. I know this is just moving existing code though, just mentioning it since I found nothing else. Looks good to me.

Reply:
Done.  I switched it to just `if not os.path.isdir(path):` because that implicitly covers `exists` as well.